Lambert-W test tolerances are too stringent or test values lack precision.

Compiled with `g++-9`, we have:

```
$ ./test_lambert_w.x
Running 2 test cases...
GCC 9.2.0

Testing type f
Tolerance 2 * epsilon == 2.38419e-07
Precision 6 decimal digits, max_digits10 = 9
-exp(-1) = -0.367879450

Testing type d
Tolerance 2 * epsilon == 4.44089e-16
Precision 15 decimal digits, max_digits10 = 17
-exp(-1) = -0.36787944117144233

Testing type e
Tolerance 16 * epsilon == 1.73472e-18
Precision 18 decimal digits, max_digits10 = 21
-exp(-1) = -0.367879441171442321583

Testing type N5boost4math8concepts12real_conceptE
Tolerance 2 * epsilon == 2.1684e-19
Precision 0 decimal digits, max_digits10 = 0
-exp(-1) = -0.4

BOOST_MATH_TEST_FLOAT128 NOT defined so NO float128 tests.
Tolerance 1 * epsilon == 2.22045e-16
test_lambert_w.cpp:1035: error: in "test_range_of_double_values": difference{2.61848e-09} between lambert_w0(create_test_value<RealType>( -0.36787944117144228Q, "-0.36787944117144228", boost::integral_constant<bool, std::numeric_limits<RealType>::is_specialized && (std::numeric_limits<RealType>::radix == 2) && (std::numeric_limits<RealType>::digits <= 113) && boost::is_convertible<largest_float, RealType>::value>(), boost::integral_constant<bool, boost::is_constructible<RealType, const char*>::value>())){-0.99999998154505632} and create_test_value<RealType>( -0.99999997892657588Q, "-0.99999997892657588", boost::integral_constant<bool, std::numeric_limits<RealType>::is_specialized && (std::numeric_limits<RealType>::radix == 2) && (std::numeric_limits<RealType>::digits <= 113) && boost::is_convertible<largest_float, RealType>::value>(), boost::integral_constant<bool, boost::is_constructible<RealType, const char*>::value>()){-0.99999997892657588} exceeds 2.2204460492503131e-16
test_lambert_w.cpp:1040: error: in "test_range_of_double_values": difference{4.64453e-10} between lambert_w0(create_test_value<RealType>( -0.36787944117144222Q, "-0.36787944117144222", boost::integral_constant<bool, std::numeric_limits<RealType>::is_specialized && (std::numeric_limits<RealType>::radix == 2) && (std::numeric_limits<RealType>::digits <= 113) && boost::is_convertible<largest_float, RealType>::value>(), boost::integral_constant<bool, boost::is_constructible<RealType, const char*>::value>())){-0.9999999746548851} and create_test_value<RealType>( -0.99999997419043196Q, "-0.99999997419043196", boost::integral_constant<bool, std::numeric_limits<RealType>::is_specialized && (std::numeric_limits<RealType>::radix == 2) && (std::numeric_limits<RealType>::digits <= 113) && boost::is_convertible<largest_float, RealType>::value>(), boost::integral_constant<bool, boost::is_constructible<RealType, const char*>::value>()){-0.99999997419043196} exceeds 2.2204460492503131e-16
(std::numeric_limits<RealType>::min)() 2.2250738585072014e-308

*** 2 failures are detected in the test module "Master Test Suite"
```

The issue is that the condition number of function evaluation near these test points is tending to infinity, so any change in rounding in any part of the computation is causing large changes in the result.

--- a/test/test_lambert_w.cpp
+++ b/test/test_lambert_w.cpp
@@ -1032,15 +1032,19 @@ BOOST_AUTO_TEST_CASE( test_range_of_double_values )
     5e7 * tolerance);// diff 2.30785e-09 v 2.2204460492503131e-16
 
   // Compare with previous PB/FK computations at double precision.
+  RealType w0 = lambert_w0(BOOST_MATH_TEST_VALUE(RealType, -0.36787944117144228));
+  RealType w0_prime = boost::math::lambert_w0_prime(BOOST_MATH_TEST_VALUE(RealType, -0.36787944117144228));
   BOOST_CHECK_CLOSE_FRACTION(  // Check float_next(-exp(-1) )
-    lambert_w0(BOOST_MATH_TEST_VALUE(RealType, -0.36787944117144228)),
-    BOOST_MATH_TEST_VALUE(RealType, -0.99999997892657588),
-    tolerance); // diff 6.03558e-09 v 2.2204460492503131e-16
+    w0,
+    BOOST_MATH_TEST_VALUE(RealType, -0.9999999849621573837115797120602890516186071783122773515945338502828025975466699519609633476854139977),
+    tolerance*abs(w0_prime/w0)); // diff 6.03558e-09 v 2.2204460492503131e-16
 
+  w0 = lambert_w0(BOOST_MATH_TEST_VALUE(RealType, -0.36787944117144222));
+  w0_prime = boost::math::lambert_w0_prime(BOOST_MATH_TEST_VALUE(RealType, -0.36787944117144222));
   BOOST_CHECK_CLOSE_FRACTION(  // Check  float_next(float_next(-exp(-1) ))
-    lambert_w0(BOOST_MATH_TEST_VALUE(RealType, -0.36787944117144222)),
+    w0,
     BOOST_MATH_TEST_VALUE(RealType, -0.99999997419043196),
-    tolerance);// diff 2.30785e-09 v 2.2204460492503131e-16
+    tolerance*abs(w0_prime/w0));// diff 2.30785e-09 v 2.2204460492503131e-16
 
                // z increasingly close to singularity.
   BOOST_CHECK_CLOSE_FRACTION(lambert_w0(BOOST_MATH_TEST_VALUE(RealType, -0.36)),

--- a/test/test_lambert_w.cpp
+++ b/test/test_lambert_w.cpp
@@ -1031,20 +1031,24 @@ BOOST_AUTO_TEST_CASE( test_range_of_double_values )
     BOOST_MATH_TEST_VALUE(RealType, -0.99999997649828679),
     5e7 * tolerance);// diff 2.30785e-09 v 2.2204460492503131e-16
 
-  // Compare with previous PB/FK computations at double precision.
-  RealType w0 = lambert_w0(BOOST_MATH_TEST_VALUE(RealType, -0.36787944117144228));
-  RealType w0_prime = boost::math::lambert_w0_prime(BOOST_MATH_TEST_VALUE(RealType, -0.36787944117144228));
-  BOOST_CHECK_CLOSE_FRACTION(  // Check float_next(-exp(-1) )
+   // Compare with previous PB/FK computations at double precision.
+   using std::abs;
+   RealType x = BOOST_MATH_TEST_VALUE(RealType, -0.36787944117144228);
+   RealType w0 = lambert_w0(x);
+   RealType w0_prime = boost::math::lambert_w0_prime(x);
+   RealType mu = std::numeric_limits<RealType>::epsilon()/2;
+   BOOST_CHECK_CLOSE_FRACTION(  // Check float_next(-exp(-1) )
     w0,
     BOOST_MATH_TEST_VALUE(RealType, -0.9999999849621573837115797120602890516186071783122773515945338502828025975466699519609633476854139977),
-    tolerance*abs(w0_prime/w0)); // diff 6.03558e-09 v 2.2204460492503131e-16
+    mu*abs(x*w0_prime/w0)); // diff 6.03558e-09 v 2.2204460492503131e-16
 
-  w0 = lambert_w0(BOOST_MATH_TEST_VALUE(RealType, -0.36787944117144222));
-  w0_prime = boost::math::lambert_w0_prime(BOOST_MATH_TEST_VALUE(RealType, -0.36787944117144222));
-  BOOST_CHECK_CLOSE_FRACTION(  // Check  float_next(float_next(-exp(-1) ))
+   x = BOOST_MATH_TEST_VALUE(RealType, -0.36787944117144222);
+   w0 = lambert_w0(x);
+   w0_prime = boost::math::lambert_w0_prime(x);
+   BOOST_CHECK_CLOSE_FRACTION(  // Check  float_next(float_next(-exp(-1) ))
     w0,
     BOOST_MATH_TEST_VALUE(RealType, -0.99999997419043196),
-    tolerance*abs(w0_prime/w0));// diff 2.30785e-09 v 2.2204460492503131e-16
+    mu*abs(x*w0_prime/w0));// diff 2.30785e-09 v 2.2204460492503131e-16
 
                // z increasingly close to singularity.
   BOOST_CHECK_CLOSE_FRACTION(lambert_w0(BOOST_MATH_TEST_VALUE(RealType, -0.36)),

--- a/test/test_lambert_w.cpp
+++ b/test/test_lambert_w.cpp
@@ -1040,7 +1040,7 @@ BOOST_AUTO_TEST_CASE( test_range_of_double_values )
    BOOST_CHECK_CLOSE_FRACTION(  // Check float_next(-exp(-1) )
     w0,
     BOOST_MATH_TEST_VALUE(RealType, -0.9999999849621573837115797120602890516186071783122773515945338502828025975466699519609633476854139977),
-    mu*abs(x*w0_prime/w0)); // diff 6.03558e-09 v 2.2204460492503131e-16
+    2*mu*abs(x*w0_prime/w0)); // diff 6.03558e-09 v 2.2204460492503131e-16
 
    x = BOOST_MATH_TEST_VALUE(RealType, -0.36787944117144222);
    w0 = lambert_w0(x);
@@ -1048,7 +1048,7 @@ BOOST_AUTO_TEST_CASE( test_range_of_double_values )
    BOOST_CHECK_CLOSE_FRACTION(  // Check  float_next(float_next(-exp(-1) ))
     w0,
     BOOST_MATH_TEST_VALUE(RealType, -0.99999997419043196),
-    mu*abs(x*w0_prime/w0));// diff 2.30785e-09 v 2.2204460492503131e-16
+    2*mu*abs(x*w0_prime/w0));// diff 2.30785e-09 v 2.2204460492503131e-16
 
                // z increasingly close to singularity.
   BOOST_CHECK_CLOSE_FRACTION(lambert_w0(BOOST_MATH_TEST_VALUE(RealType, -0.36)),

